src/IO/S3/copyS3File.cpp

```diff
@@ -269,9 +265,6 @@ namespace
             {
                 for (size_t part_number = 1; position < end_position; ++part_number)
                 {
-                    if (multipart_upload_aborted)
-                        break; /// No more part uploads.
```

Comment:
This check is redundant because `uploadPart` throws, thus execution does not reach this code path.

Reply:
`uploadPart` throws only exception which happens on schedule. It does not propagate exceptions from asynchronous task.